### PR TITLE
Pricing oracles

### LINKED ISSUE #83: Improve ad-hoc pricing solver user experience
- [x] As a pricing callback

### LINKED ISSUE #206: Mapping Coluna Vars/Constrs <-> JuMP Vars/Constrs
Needed for :
- #83 
- defining branching strategies
- cuts callback interface

### PR DESCRIPTION
**Pull request by guimarqu**
Fix #83 
Fix #206 : map is already there for variables (field `varmap` of `Optimizer`)

### CODE REVIEW

File: src/MOIcallbacks.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+)
+    form = cb.callback_data.form 
+    S = getobjsense(form)
+    result = OptimizationResult{S}()
+    pb = PrimalBound(form, cost)
+
+    colunavarids = [_get_orig_varid_in_form(model, form, v) for v in variables]
+
+    # setup variable
+    setup_var_id = [id for (id,v) in Iterators.filter(
```
Comment (vitornesello) [inline src/MOIcallbacks.jl]: isn't it very slow? to filter all the variables to get only one of them?

File: src/MOIcallbacks.jl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    uid = getuid(callback_data.form)
+    axis_index_value = model.annotations.ann_per_form[uid].axis_index_value
+    return axis_index_value
+end
+
+function MOI.get(
+    model::Optimizer, vc::BD.PricingVariableCost{MP.PricingCallbackData}, 
+    x::MOI.VariableIndex
+)
+    form = vc.callback_data.form
```
Comment (vitornesello) [inline src/MOIcallbacks.jl]: why vc here? shouldnt it be var?

File: src/MOIcallbacks.jl [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+function MOI.get(model::Optimizer, spid::BD.PricingSubproblemId{MP.PricingCallbackData})
+    callback_data = spid.callback_data
+    uid = getuid(callback_data.form)
+    axis_index_value = model.annotations.ann_per_form[uid].axis_index_value
+    return axis_index_value
+end
+
+function MOI.get(
+    model::Optimizer, vc::BD.PricingVariableCost{MP.PricingCallbackData},
```
Comment (vitornesello) [inline src/MOIcallbacks.jl]: why do we need `BD.PricingVariableCost{MP.PricingCallbackData}`. cant it be only `BD.VariableCost{MP.PricingCallbackData}`?

Comment (vitornesello) [reply src/MOIcallbacks.jl]: or `BD.VariablePrimal{MP.PricingCallbackData}`

File: src/MOIcallbacks.jl (hunk 1 above)
Comment (guimarqu) [reply src/MOIcallbacks.jl]: We have nothing to select only this variable. Filtering all variables is the only way to get it for now.

File: src/MOIcallbacks.jl (hunk 2 above)
Comment (guimarqu) [reply src/MOIcallbacks.jl]: I'll rename `pvc`

File: src/MOIcallbacks.jl (hunk 3 above)
Comment (guimarqu) [reply src/MOIcallbacks.jl]: we can change the name but it will be for a future release of BlockDecomposition.

Comment (guimarqu) [reply src/MOIcallbacks.jl]: I'll try with `MOI.VariablePrimal`

File: src/MOIcallbacks.jl (hunk 1 above)
Comment (vitornesello) [reply src/MOIcallbacks.jl]: ok. just something to keep in mind

**Pull request merged**